app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -0,0 +1,422 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/widgets/default_button.dart';
+import 'package:acter/common/widgets/search.dart';
+import 'package:acter/common/widgets/side_sheet.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter/features/home/widgets/space_chip.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+// ChildRoomType configures the sub child type of the `Spaces`
+enum ChildRoomType {
+  chat,
+  space,
+  recommendedSpace,
+}
+
+class LinkRoomPage extends ConsumerStatefulWidget {
+  final String? parentSpaceId;
+  final String? childRoomType;
```

Comment:
I'd say the if-else in the initState below shouldn't happen here, but instead this should already be typed inputs and the routing controller should determine those:


```suggestion
  final String parentSpaceId;
  final ChildRoomType childRoomType;
```